templates/tapestry-d3-vue/src/components/RootNodeButton.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <div id="root-node-container">
+    <div id="root-node-btn" @click="showModal">
+      <i class="fas fa-plus-circle fa-5x"></i>
+      <div id="root-node-label">Add Root Node</div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'root-node-button',
+  data() {
+    return {}
+  },
+  methods: {
+    showModal() {
+      console.log('here')
+      // Populate title
+      $('#createNewNodeModalLabel').text("Add root node")
+      $("#submit-add-new-node").hide()
+      $("#submit-edit-node").hide()
+      $("#submit-add-root-node").show()
+      $("#appearsat-section").hide()
+
+      // Show the modal
+      $("#createNewNodeModal").modal()
+    }
+  }
+}
+</script>
+
+<style scoped>
+#root-node-btn {
```

Comment:
If you absolutely have to use IDs, please prefix all of them with`tapestry-`. I recommend having one wrapper with an ID and then use classes for the nested elements. That way, you can do something like `#tapestry-add-root-node-wrapper > .btn`, `#tapestry-add-root-node-wrapper > label`, etc.

Reply:
`#root-node-btn` is now also used for styling in tapestry.css. Could we leave this for later as well?

Reply:
Yes, you can leave this for later - just make sure you make a list of the things that still need to be done.